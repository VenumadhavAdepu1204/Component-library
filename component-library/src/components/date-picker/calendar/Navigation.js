import React, { useEffect, useState } from 'react'
import { NavigationWrapper, LeftArrow, NavigationLabel, RightArrow } from '../styles'
import { getMonthYearLabel, getYearLabel, abbreviationForMonth, months } from '../shared/helpers'
import Icon from '../../../../IB/components/icon'
import styleVars from '../../../../common/styles-global/global.js'

const Navigation = props => {
  const { drillUp, onLeftArrowClick, onRightArrowClick, fullDate, view, year, month, currentDecadeValue } = props

  const [decadeValue, setDecadeValue] = useState(currentDecadeValue)
  const [monthNumber, setMonthNumber] = useState(month)
  const [yearNumber, setYearNumber] = useState(year)

  const nextLabel = <Icon iconName='Arrow Right' color={styleVars.color.ocean_blue} />
  const prevLabel = <Icon iconName='Arrow Left' color={styleVars.color.ocean_blue} />

  useEffect(() => {
    setDecadeValue(currentDecadeValue)
    setYearNumber(year)
    setMonthNumber(month)
  }, [currentDecadeValue], [month], [year])

  const getLabel = (fullDate, view) => {
    switch (view) {
      case 'decade': return decadeValue
      case 'year': return (yearNumber !== null) ? yearNumber : getYearLabel(fullDate)
      case 'month': return (monthNumber !== null) ? abbreviationForMonth(months[monthNumber]) + ' ' + yearNumber : getMonthYearLabel(fullDate)
      default: break
    }
  }

  return (
    <NavigationWrapper>
      <LeftArrow data-test-id='left-arrow' onClick={onLeftArrowClick}>{prevLabel}</LeftArrow>
      <NavigationLabel data-test-id='navigation-label' onClick={drillUp}>{getLabel(fullDate, view)}</NavigationLabel>
      <RightArrow data-test-id='right-arrow' onClick={onRightArrowClick}>{nextLabel}</RightArrow>
    </NavigationWrapper>
  )
}

export default Navigation
