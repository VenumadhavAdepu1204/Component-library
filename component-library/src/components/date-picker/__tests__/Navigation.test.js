import React from 'react'
import { render, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Navigation from '../calendar/Navigation'

configure({ adapter: new Adapter() })

const app = render(<Navigation
  fullDate={new Date(2018, 2)}
  drillUp={jest.fn()}
  month='3'
  year='2019'
  currentDecadeValue='2009-2019'
  onLeftArrowClick={jest.fn()}
  onRightArrowClick={jest.fn()}
  view='month'
/>)

test('renders properly', () => {
  expect(app).toMatchSnapshot()
})
