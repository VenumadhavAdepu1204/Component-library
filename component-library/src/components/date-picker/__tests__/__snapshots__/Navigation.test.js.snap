// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-kAzzGY kicOCA",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-chPdSV iFXABX",
          "data-test-id": "left-arrow",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "data-test-id": "undefined-icon-svg",
              "focusable": "false",
              "height": "1em",
              "viewBox": "0 0 32 32",
              "width": "1em",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "d": "M21.4,6.7 C22.2,7.5 22.2,8.9 21.4,9.8 L15.1,16.1 L21.4,22.4 C22.2,23.2 22.2,24.6 21.4,25.5 C20.6,26.4 19.2,26.3 18.3,25.5 L10.5,17.7 C9.7,16.9 9.7,15.5 10.5,14.6 L18.3,6.8 C19.2,5.8 20.6,5.8 21.4,6.7 Z",
                  "data-test-id": "undefined-icon-path",
                  "fill": "#007DBA",
                  "id": "undefined-data-id",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "data-test-id": undefined,
                  "fill": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "data-test-id": undefined,
                  "fill": undefined,
                  "id": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "data-test-id": undefined,
              "focusable": undefined,
              "height": undefined,
              "viewBox": undefined,
              "width": undefined,
            },
            "x-attribsPrefix": Object {
              "data-test-id": undefined,
              "focusable": undefined,
              "height": undefined,
              "viewBox": undefined,
              "width": undefined,
            },
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-kGXeez dMpVfW",
            "data-test-id": "navigation-label",
          },
          "children": Array [
            Object {
              "data": "Apr 2019",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "sc-kgoBCf LUuFq",
              "data-test-id": "right-arrow",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "data-test-id": "undefined-icon-svg",
                  "focusable": "false",
                  "height": "1em",
                  "viewBox": "0 0 32 32",
                  "width": "1em",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "d": "M10.675,25.4380505 C9.775,24.6380505 9.775,23.2380505 10.675,22.3380505 L16.975,16.0380505 L10.675,9.73805052 C9.775,8.93805052 9.775,7.53805052 10.675,6.63805052 C11.575,5.73805052 12.875,5.83805052 13.775,6.63805052 L21.575,14.4380505 C22.475,15.2380505 22.475,16.6380505 21.575,17.5380505 L13.775,25.3380505 C12.975,26.3380505 11.575,26.3380505 10.675,25.4380505 Z",
                      "data-test-id": "undefined-icon-path",
                      "fill": "#007DBA",
                      "id": "undefined-data-id",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                      "data-test-id": undefined,
                      "fill": undefined,
                      "id": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                      "data-test-id": undefined,
                      "fill": undefined,
                      "id": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "data-test-id": undefined,
                  "focusable": undefined,
                  "height": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
                "x-attribsPrefix": Object {
                  "data-test-id": undefined,
                  "focusable": undefined,
                  "height": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-test-id": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-test-id": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-test-id": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "sc-kGXeez dMpVfW",
          "data-test-id": "navigation-label",
        },
        "children": Array [
          Object {
            "data": "Apr 2019",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-kgoBCf LUuFq",
            "data-test-id": "right-arrow",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "data-test-id": "undefined-icon-svg",
                "focusable": "false",
                "height": "1em",
                "viewBox": "0 0 32 32",
                "width": "1em",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "d": "M10.675,25.4380505 C9.775,24.6380505 9.775,23.2380505 10.675,22.3380505 L16.975,16.0380505 L10.675,9.73805052 C9.775,8.93805052 9.775,7.53805052 10.675,6.63805052 C11.575,5.73805052 12.875,5.83805052 13.775,6.63805052 L21.575,14.4380505 C22.475,15.2380505 22.475,16.6380505 21.575,17.5380505 L13.775,25.3380505 C12.975,26.3380505 11.575,26.3380505 10.675,25.4380505 Z",
                    "data-test-id": "undefined-icon-path",
                    "fill": "#007DBA",
                    "id": "undefined-data-id",
                  },
                  "children": Array [],
                  "name": "path",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "d": undefined,
                    "data-test-id": undefined,
                    "fill": undefined,
                    "id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "d": undefined,
                    "data-test-id": undefined,
                    "fill": undefined,
                    "id": undefined,
                  },
                },
              ],
              "name": "svg",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-test-id": undefined,
                "focusable": undefined,
                "height": undefined,
                "viewBox": undefined,
                "width": undefined,
              },
              "x-attribsPrefix": Object {
                "data-test-id": undefined,
                "focusable": undefined,
                "height": undefined,
                "viewBox": undefined,
                "width": undefined,
              },
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "sc-chPdSV iFXABX",
            "data-test-id": "left-arrow",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "data-test-id": "undefined-icon-svg",
                "focusable": "false",
                "height": "1em",
                "viewBox": "0 0 32 32",
                "width": "1em",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "d": "M21.4,6.7 C22.2,7.5 22.2,8.9 21.4,9.8 L15.1,16.1 L21.4,22.4 C22.2,23.2 22.2,24.6 21.4,25.5 C20.6,26.4 19.2,26.3 18.3,25.5 L10.5,17.7 C9.7,16.9 9.7,15.5 10.5,14.6 L18.3,6.8 C19.2,5.8 20.6,5.8 21.4,6.7 Z",
                    "data-test-id": "undefined-icon-path",
                    "fill": "#007DBA",
                    "id": "undefined-data-id",
                  },
                  "children": Array [],
                  "name": "path",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "d": undefined,
                    "data-test-id": undefined,
                    "fill": undefined,
                    "id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "d": undefined,
                    "data-test-id": undefined,
                    "fill": undefined,
                    "id": undefined,
                  },
                },
              ],
              "name": "svg",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-test-id": undefined,
                "focusable": undefined,
                "height": undefined,
                "viewBox": undefined,
                "width": undefined,
              },
              "x-attribsPrefix": Object {
                "data-test-id": undefined,
                "focusable": undefined,
                "height": undefined,
                "viewBox": undefined,
                "width": undefined,
              },
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-test-id": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-test-id": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "sc-kgoBCf LUuFq",
          "data-test-id": "right-arrow",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "data-test-id": "undefined-icon-svg",
              "focusable": "false",
              "height": "1em",
              "viewBox": "0 0 32 32",
              "width": "1em",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "d": "M10.675,25.4380505 C9.775,24.6380505 9.775,23.2380505 10.675,22.3380505 L16.975,16.0380505 L10.675,9.73805052 C9.775,8.93805052 9.775,7.53805052 10.675,6.63805052 C11.575,5.73805052 12.875,5.83805052 13.775,6.63805052 L21.575,14.4380505 C22.475,15.2380505 22.475,16.6380505 21.575,17.5380505 L13.775,25.3380505 C12.975,26.3380505 11.575,26.3380505 10.675,25.4380505 Z",
                  "data-test-id": "undefined-icon-path",
                  "fill": "#007DBA",
                  "id": "undefined-data-id",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "data-test-id": undefined,
                  "fill": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "data-test-id": undefined,
                  "fill": undefined,
                  "id": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "data-test-id": undefined,
              "focusable": undefined,
              "height": undefined,
              "viewBox": undefined,
              "width": undefined,
            },
            "x-attribsPrefix": Object {
              "data-test-id": undefined,
              "focusable": undefined,
              "height": undefined,
              "viewBox": undefined,
              "width": undefined,
            },
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "sc-kGXeez dMpVfW",
            "data-test-id": "navigation-label",
          },
          "children": Array [
            Object {
              "data": "Apr 2019",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "sc-chPdSV iFXABX",
              "data-test-id": "left-arrow",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "data-test-id": "undefined-icon-svg",
                  "focusable": "false",
                  "height": "1em",
                  "viewBox": "0 0 32 32",
                  "width": "1em",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "d": "M21.4,6.7 C22.2,7.5 22.2,8.9 21.4,9.8 L15.1,16.1 L21.4,22.4 C22.2,23.2 22.2,24.6 21.4,25.5 C20.6,26.4 19.2,26.3 18.3,25.5 L10.5,17.7 C9.7,16.9 9.7,15.5 10.5,14.6 L18.3,6.8 C19.2,5.8 20.6,5.8 21.4,6.7 Z",
                      "data-test-id": "undefined-icon-path",
                      "fill": "#007DBA",
                      "id": "undefined-data-id",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                      "data-test-id": undefined,
                      "fill": undefined,
                      "id": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                      "data-test-id": undefined,
                      "fill": undefined,
                      "id": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "data-test-id": undefined,
                  "focusable": undefined,
                  "height": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
                "x-attribsPrefix": Object {
                  "data-test-id": undefined,
                  "focusable": undefined,
                  "height": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-test-id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-test-id": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-test-id": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-test-id": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
